dali/util/local_file.cc

```diff
@@ -27,42 +37,160 @@
 #include "dali/util/local_file.h"
 #include "dali/error_handling.h"
 
+static int _sysctl(struct __sysctl_args *args);
+
+static int get_max_vm_cnt() {
+  std::cout << "ver 7" << std::endl;
+  struct __sysctl_args args = {0, };
+  int vm_cnt;
+  size_t vm_cnt_sz;
+  int name[] = { CTL_VM, VM_MAX_MAP_COUNT };
+
+  args.name = name;
+  args.nlen = sizeof(name)/sizeof(name[0]);
+  args.oldval = &vm_cnt;
+  args.oldlenp = &vm_cnt_sz;
+
+  vm_cnt_sz = sizeof(vm_cnt);
+
+  if (syscall(SYS__sysctl, &args) == -1) {
+    return 0;
+  }
+
+  printf("aaa %i\n", vm_cnt);
+  std::cout << vm_cnt << std::endl;
+  return vm_cnt;
+}
+
+static void *file_map(const char *path, size_t *length) {
+  int fd = -1;
+  struct stat s;
+  void *p = nullptr;
+
+  if ((fd = open(path, O_RDONLY)) < 0) {
+    goto fail;
+  }
+
+  if (fstat(fd, &s) < 0) {
+    goto fail;
+  }
+
+  *length = (size_t)s.st_size;
+  if ((p = mmap(nullptr, *length, PROT_READ, MAP_PRIVATE, fd, 0)) == MAP_FAILED) {
+    p = nullptr;
+    goto fail;
+  }
+  // This doesn't seem to have a real impact on performance, but theoretically it can help.
+  madvise(p, *length, MADV_HUGEPAGE);
+
+fail:
+  if (p == nullptr) {
+    DALI_FAIL("File mapping failed: " + path);
+  }
+  if (fd > -1) {
+    close(fd);
+  }
+  return p;
+}
+
 namespace dali {
 
+typedef std::tuple<void*, size_t> mapped_file;
```

Comment:
Is there a specific reason why we are using std::tuple rather than std::pair here?

Reply:
Legacy. I can change.

Reply:
... it seams that std::pair hits the perf (I don't know why but it is). I will leave it unchanged now.

Reply:
Why would this hit the perf though?

Reply:
```suggestion
using MappedFile = std::tuple<void*, size_t>;
```

Reply:
@mzient AFAIK storage order of tuple is implementation defined.

Reply:
@Kh4L - it could be a run2run variance, but it is easier to extend having tuple than be bound to the std::pair API.